### DISCUSSION TITLE
Semi-random NetworkPolicy violations afer upgrade to v1.27.7 ?

### DISCUSSION THREAD

**Discussion by ondrejmo** [k3s-io/k3s#8763] [Q&A]
After I updated from v1.27.6 to v1.27.7 yesterday I started to see semi-random violations of almost all NetworkPolicies in the cluster. It seem to be non-deterministic, because some connections (from/to the same host/port) pass and others are blocked. No other changes occured in the cluster in the 24 hours before the update. Does anyone else experience this issue?

![image](https://github.com/k3s-io/k3s/assets/11840034/a2273451-52bd-4835-9126-c3f3b97455a9)

Just to illustrate the amount and contrast with the previous state.

![image](https://github.com/k3s-io/k3s/assets/11840034/d2d27069-9bac-4e81-8c69-9c1cef6ba2a9)

Edit: the manual downgrade to 1.27.6 removes this issue

**Comment by ondrejmo** [comment]
Unfortunately the issue seems to persist even in `v1.27.8+k3s2` and `v1.28.4+k3s2`, rollback to `v1.27.6` still fixes the issue.

**Comment by brandond** [comment]
You didn't open an issue and didn't offer any details so there's not much for us to work with. What Linux distro are your nodes running? What iptables version? Are you using the default k3s CNI and network policy engine, or something else? How are you measuring these events?

**Reply by ondrejmo** [reply]
Sorry, initially, I was hoping for it to be some obvious regression, but it seems that it only affects my setup. https://github.com/k3s-io/k3s/issues/9032
